be/src/exec/jni_scanner.cpp

```diff
@@ -420,4 +496,141 @@ Status HiveJniScanner::do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk)
     return status;
 }
 
+JniScanner* create_hive_jni_scanner(const JniScanner::CreateOptions& options) {
+    const auto& scan_range = *(options.scan_range);
+    const HiveTableDescriptor* hive_table = options.hive_table;
+    static const char* serde_property_prefix = "SerDe.";
+
+    std::string data_file_path;
+    std::string hive_column_names;
+    std::string hive_column_types;
+    std::string serde;
+    std::string input_format;
+    std::map<std::string, std::string> serde_properties;
+    std::string time_zone;
+
+    if (dynamic_cast<const FileTableDescriptor*>(hive_table)) {
+        const auto* file_table = dynamic_cast<const FileTableDescriptor*>(hive_table);
+
+        data_file_path = scan_range.full_path;
+
+        hive_column_names = file_table->get_hive_column_names();
+        hive_column_types = file_table->get_hive_column_types();
+        serde = file_table->get_serde_lib();
+        input_format = file_table->get_input_format();
+        time_zone = file_table->get_time_zone();
+    } else {
+        const auto* hdfs_table = dynamic_cast<const HdfsTableDescriptor*>(hive_table);
+
+        auto* partition_desc = hdfs_table->get_partition(scan_range.partition_id);
+        std::string partition_full_path = partition_desc->location();
+        data_file_path = fmt::format("{}/{}", partition_full_path, scan_range.relative_path);
+
+        hive_column_names = hdfs_table->get_hive_column_names();
+        hive_column_types = hdfs_table->get_hive_column_types();
+        serde = hdfs_table->get_serde_lib();
+        input_format = hdfs_table->get_input_format();
+        serde_properties = hdfs_table->get_serde_properties();
+        time_zone = hdfs_table->get_time_zone();
+    }
+
+    std::map<std::string, std::string> jni_scanner_params;
+
+    jni_scanner_params["hive_column_names"] = hive_column_names;
+    jni_scanner_params["hive_column_types"] = hive_column_types;
+    jni_scanner_params["data_file_path"] = data_file_path;
+    jni_scanner_params["block_offset"] = std::to_string(scan_range.offset);
+    jni_scanner_params["block_length"] = std::to_string(scan_range.length);
+    jni_scanner_params["serde"] = serde;
+    jni_scanner_params["input_format"] = input_format;
+    jni_scanner_params["time_zone"] = time_zone;
+    jni_scanner_params["fs_options_props"] = build_fs_options_properties(*(options.fs_options));
+
+    for (const auto& pair : serde_properties) {
+        jni_scanner_params[serde_property_prefix + pair.first] = pair.second;
+    }
+
+    std::string scanner_factory_class = "com/starrocks/hive/reader/HiveScannerFactory";
+
+    return new HiveJniScanner(scanner_factory_class, jni_scanner_params);
```

Comment:
return `unique_ptr<JniScanner>` is better, just in case the caller call the interface without processing the return value, the memory can be freed correctly.

Reply:
Well, in caller side, this instance be will added to ObjectPool.  So we can not return unique_ptr

![image](https://github.com/StarRocks/starrocks/assets/1081215/263115dc-9251-4c5f-a2b2-5410f485ca92)

Reply:
Well.  if you insist.